### PR TITLE
fix(programs/psa): commit python3 generated file

### LINKED ISSUE #3524: Cannot open include file: 'psa_constant_names_generated.c': No such file or directory
- Type: Bug
- Priority: Blocker | Major

---------------------------------------------------------------
## Bug

**OS**  
Mbed OS|linux|windows|

**mbed TLS build:**  
Version: 2.23.0

**Expected behavior** 
After taking the latest code and targeting the code to use updated build tools. I was expecting it to run after I build using (ctrl+shift+B).

**Actual behavior**  
I am facing an error after the build(ctrl+shift+B).

**Severity	Code	Description	Project	File	Line	Suppression State
Error	C1083	Cannot open include file: 'psa_constant_names_generated.c': No such file or directory	psa_constant_names	C:\Users\U376361\Documents\Visual Studio 2017\Projects\mbedtls-mbedtls-2.23.0\programs\psa\psa_constant_names.c	142**	

**Steps to reproduce**  
1. Just download the latest version 2.23.0.
2. Open visual studio 2019.
3. Open mbedtls solution file.
4. Retarget solution
5. After success, run the command **ctrl+shift+B**  while keeping the solution configuration as the release.
image

### PR DESCRIPTION
**Pull request by CameronNemo**
Fixes #3524

Signed-off-by: Cameron Nemo <cnemo@tutanota.com>

## Description

Python should not be required for the build when the no_test target is
used. This commit adds the generated file to the source tree and the
check-generated-files script, and removes the generation from (c)make.

## Status

READY

## Requires Backporting

NO

## Migrations

NO

## Todos

- [x] Tests UNNEEDED
- [x] Documentation UNNEEDED
- [x] Changelog updated
- [x] Backported UNNEEDED

## Steps to test or reproduce

Build with `make no_test` and without python3.

### CODE REVIEW

**Comment by CameronNemo** [comment]
Looks like Windows does not like the call to os.rename. Anyone with a Windows license and/or expertise have a recommendation?

```pytb
Traceback (most recent call last):
  File "scripts/generate_psa_constants.py", line 423, in <module>
    OUTPUT_FILE_DIR + '/psa_constant_names_generated.c')
  File "scripts/generate_psa_constants.py", line 414, in generate_psa_constants
    os.rename(temp_file_name, output_file_name)
FileExistsError: [WinError 183] Cannot create a file when that file already exists: 'programs/psa/psa_constant_names_generated.c.tmp' -> 'programs/psa/psa_constant_names_generated.c'
```

**Comment by ronald-cron-arm** [comment]
Small question how do you "Build with make no_test and without python3."? Would it be possible to add a non-regression test based on that? If we don't have a regression test it is likely that some Python dependency will sneak in `make no_test` at some point in the future.

**Comment by ronald-cron-arm** [comment]
Otherwise the CI build Windows-mingw fails. Here is the log:

C:\builds\ws\-3709-head-DXGVNTBUK6QRR67RSZNDG>python windows_testing.py src logs -b mingw 
2020-09-22 22:56:10,267 - MinGW - INFO - Checking out git worktree
2020-09-22 22:56:10,895 - MinGW - INFO - Preparing C:/builds/ws/-3709-head-DXGVNTBUK6QRR67RSZNDG/worktrees/tmptdvg3igb (identifier tmptdvg3igb)
HEAD is now at 5c8f5f1fb fix(programs/psa): commit python3 generated file
2020-09-22 22:56:11,170 - MinGW - INFO - 
2020-09-22 22:56:11,170 - MinGW - INFO - Enabling as much of scripts\config.pl as possible
2020-09-22 22:56:11,247 - MinGW - INFO - scripts\config.pl: python3: No such file or directory. Trying python instead.
2020-09-22 22:56:11,326 - MinGW - INFO - scripts\config.pl: python3: No such file or directory. Trying python instead.
2020-09-22 22:56:11,403 - MinGW - INFO - scripts\config.pl: python3: No such file or directory. Trying python instead.
2020-09-22 22:56:11,481 - MinGW - INFO - scripts\config.pl: python3: No such file or directory. Trying python instead.
2020-09-22 22:56:11,559 - MinGW - INFO - scripts\config.pl: python3: No such file or directory. Trying python instead.
2020-09-22 22:56:11,637 - MinGW - INFO - scripts\config.pl: python3: No such file or directory. Trying python instead.
2020-09-22 22:56:11,715 - MinGW - INFO - scripts\config.pl: python3: No such file or directory. Trying python instead.
2020-09-22 22:56:11,792 - MinGW - INFO - scripts\config.pl: python3: No such file or directory. Trying python instead.
2020-09-22 22:56:11,798 - MinGW - INFO - Building mbed TLS using mingw32-make
2020-09-22 22:56:11,888 - MinGW - ERROR - Could Not Find C:\builds\ws\-3709-head-DXGVNTBUK6QRR67RSZNDG\worktrees\tmptdvg3igb\library\del_errors_out_if_the_file_list_is_empty_but_not_if_a_file_does_not_exist
Could Not Find C:\builds\ws\-3709-head-DXGVNTBUK6QRR67RSZNDG\worktrees\tmptdvg3igb\3rdparty\everest\library\everest.o
The syntax of the command is incorrect.
Makefile:364: recipe for target 'clean' failed
mingw32-make[1]: *** [clean] Error 1
Makefile:89: recipe for target 'clean' failed
mingw32-make: *** [clean] Error 2
2020-09-22 22:56:12,126 - MinGW - INFO - 
MingW build failed
1 configurations tested, 0 successful
script returned exit code 2

File: programs/Makefile [hunk 1]
```diff
@@ -371,12 +359,11 @@ clean:
 ifndef WINDOWS
 	rm -f $(APPS)
 	-rm -f ssl/ssl_pthread_server$(EXEXT)
-	rm -f $(EXTRA_GENERATED)
 	-rm -f test/cpp_dummy_build$(EXEXT)
 else
 	if exist *.o del /Q /F *.o
 	if exist *.exe del /Q /F *.exe
-	del /S /Q /F $(EXTRA_GENERATED)
+	del /S /Q /F
```
Comment (gilles-peskine-arm) [inline programs/Makefile:366]: This line needs to be removed altogether.

**Review by gilles-peskine-arm — CHANGES_REQUESTED** [review]
Thanks! Other than my two comments, look good to me.

**Comment by gilles-peskine-arm** [comment]
> Would it be possible to add a non-regression test based on that? If we don't have a regression test it is likely that some Python dependency will sneak in make no_test at some point in the future.

@ronald-cron-arm Yes but that's outside the scope of this PR. It would be good to improve our out-of-box testing to ensure that it isn't relying on some tool that we don't document as a dependency, but we currently don't have that. I've filed https://github.com/ARMmbed/mbedtls/issues/3714 and https://github.com/ARMmbed/mbedtls/issues/3713.

**Comment by ronald-cron-arm** [comment]
> > Would it be possible to add a non-regression test based on that? If we don't have a regression test it is likely that some Python dependency will sneak in make no_test at some point in the future.
> 
> @ronald-cron-arm Yes but that's outside the scope of this PR. It would be good to improve our out-of-box testing to ensure that it isn't relying on some tool that we don't document as a dependency, but we currently don't have that. I've filed #3714 and #3713.

ok, given that `no_test` is not run as part of the CI it makes sense to me add some testing later.

File: programs/Makefile (hunk 1 above)
Comment (CameronNemo) [reply programs/Makefile:366]: oh these are flags huh! never been on a DOS or NT CLI. I will get this fixed.

**Review by gilles-peskine-arm — APPROVED** [review]

**Comment by ronald-cron-arm** [comment]
Thanks a lot for this. LGTM as well.

**Review by ronald-cron-arm — APPROVED** [review]

**Pull request merged**